### ISSUE TITLE
centreon_hp_procurve.pl --plugin=network::hp::procurve::snmp::plugin --mode=virtual-chassis 

### ISSUE THREAD

**Issue by Didier97180** [centreon/centreon-plugins#4001]
Hello,

Regarding the plugin centreon_hp_procurve.pl --plugin=network::hp::procurve::snmp::plugin --mode=virtual-chassis

Regarding the stack there are two links per Switch. The first operational and not the second, but centreon takes the second into account and gives me Critical status.

I would like if possible that it does not take into consideration the second link.
![image](https://user-images.githubusercontent.com/113202138/196485065-8aa2b953-b6c8-4b17-a188-8c2a8878344d.png)

**Comment by Sims24** [comment]
Hi, 

In your case, you can modify the default value of the --critical-link-status option. 

Adding `--critical-link-status='%{display} !~ /I-Link(1|2)_2/ && %{link_status} eq "down"' ` will do what you want I think.

**Comment by Didier97180** [comment]
Hello,

Very cool it worked.
On the other hand, I'm going to be annoying, I would like to know the meaning of the command that allows you to do the action. I would like to understand and not settle for the answer :)

**Comment by Sims24** [comment]
Great, you're not annoying, it's important to understand the why, as this applies to every probe checking status strings.

The first thing to do is displaying the help of a mode: 

`./centreon_hp_procurve.pl --plugin=network::hp::procurve::snmp::plugin --mode=virtual-chassis --help`

At the end of the output you will find options specific to this mode (the interesting part below). In this situation you want to override the default status by a custom one, you should have identified that it is related to your vc links: 

```
    --unknown-link-status
            Set unknown threshold for status. Can used special variables
            like: %{link_status}, %{display}

    --warning-link-status
            Set warning threshold for status. Can used special variables
            like: %{link_status}, %{display}

    --critical-link-status
            Set critical threshold for status (Default: '%{link_status} eq
            "down"'). Can used special variables like: %{link_status},
            %{display}
```

Two interesting bits of information in the output above: 
- there is a default value set for `--critical-link-status` which is `'%{link_status} eq "down"'
- you can use those placeholders (it's our convention) to work with scrapped data: %{display} which holds the name/instance of the link, and %{link_status} which holds the actual status of a link (e.g up, down, ...).

Then, I tuned the value of --critical-link-status to have a display matching the two secondary links of the vc and evaluate %{display} against it: `%{display} !~ /I-Link(1|2)_2/` matches both 'I-Link1_2' and 'I-Link1_2'. I keep the critical alert only when the status is down. (`'%{link_status} eq "down"').

So, basically, `--critical-link-status='%{display} !~ /I-Link(1|2)_2/ && %{link_status} eq "down"'` can be translated in human friendly form like this: 
- Trigger a critical alert when the status of the link is sown and the name/instance of the link isn't equal to  'I-Link1_2' or 'I-Link1_2'. 

Hope this helps

**Comment by Didier97180** [comment]
Great, thanks for the explanation?
However, where I didn't understand is when you say "Then, I tuned the value of --critical-link-status to have a display matching the two secondary links of the vc and evaluate %{display} against it: %{display} !~ /I-Link(1|2)_2/ matches both 'I-Link1_2' and 'I-Link1_2'"  . wouldn't it be both 'I-Link1_2' and 'I-Link2_2'.

  So in human-friendly form it would be rather than "  Trigger a critical alert when the status of the link is sown and the name/instance of the link isn't equal to 'I-Link1_2' or 'I-Link2_2'".

maybe i misunderstood

**Comment by Sims24** [comment]
Indeed it's a copy paste issue. You fixed it right.

**Comment by Didier97180** [comment]
34 / 5 000
Résultats de traduction
It proves that I understand correctly :)
Anyway, thank you very much for your time.
